inst/htmlwidgets/datatables.js

```diff
@@ -257,15 +257,28 @@ HTMLWidgets.widget({
       options.ajax.dataSrc = function(json) {
         DT_rows_all = $.makeArray(json.DT_rows_all);
         DT_rows_current = $.makeArray(json.DT_rows_current);
-        return json.data;
+        var data = json.data;
+        if (!colReorderEnabled()) return data;
+        var table = $table.DataTable(), order = table.colReorder.order(), flag = true, i, j, row;
+        for (i = 0; i < order.length; ++i) if (order[i] !== i) flag = false;
+        if (flag) return data;
+        for (i = 0; i < data.length; ++i) {
+          row = data[i].slice();
+          for (j = 0; j < order.length; ++j) data[i][j] = row[order[j]];
+        }
```

Comment:
I wonder if there is a way to reorder the data on the server side. If there is, the implementation could be much easier (`data[order]` in R instead of a nested loop in JS). Of course, it depends on whether we have the order information on the server side.

Reply:
In that case, let's just forget it. I'm not too concerned about the performance of the nested loops here since `data` should typically be small enough.

Reply:
I had the concern too... For the deep copy and nested loops... but as you said, it should be fine in the most cases.